### DISCUSSION TITLE
Using Okta Provider after v3.20

### DISCUSSION THREAD

**Discussion by slapner** [nextauthjs/next-auth#1287] [Help] [answered]
Has anyone been able to use Okta as a provider since the 3.2.0 release? My application was working fine before the url patterns changed in NextAuth and now I only get a 404 when I try to sign-in.

**Comment by balazsorban44** [comment, marked as answer]
https://github.com/nextauthjs/next-auth/pull/1257 could this be related, or the issue occurs in stable 3.2.0 already?

I'm that case, https://github.com/nextauthjs/next-auth/issues/1022 here is the one that could have caused issues. Do you think I overlooked something?

**Reply by slapner** [reply]
Finally just figured out that I need to append `/oauth2` to my Okta domain variable.

**Reply by slapner** [reply]
I think the change you made is correct because not all orgs use the same authorization endpoint but maybe adding a note in the documentation could help people in the future

**Reply by iaincollins** [reply]
@slapner Thanks for the insights!

Do you have any specific recommendations / example usage we could add to the documentation?

Do you think the defaults should be different from what they currently are?

It's hard to keep track of best practice with some providers,especially when there are multiple ways of using some of them (e.g. Okta, Keycloak, Azure) so any insight or links to relevant documentation or official examples are most welcome.

**Reply by slapner** [reply]
@iaincollins I think the confusion might come from the parameter being called 'domain'

![image](https://user-images.githubusercontent.com/20480013/107569978-e67b6b00-6bb6-11eb-8314-35af18552487.png)

In the Okta admin panel there is a 'Okta Domain' that you can copy with one click (like the clientId, secret) so if someone does that and doesn't add '/oauth2' (at minimun) after their domain then it will not work. Maybe just a note about checking the AuthorizationUrl pattern for their organization? Most people will just have '/oauth2' but others may have more.

**Reply by brandonchinn178** [reply]
ah! I wrote an issue for this: https://github.com/nextauthjs/next-auth/issues/1345

I think it would be helpful to rename `domain` to `baseUrl`, to correspond with Okta's documentation, as well as how `@okta/jwt-verifier` has you configure the Okta URL

e.g. I currently do
```js
domain: `${process.env.OKTA_DOMAIN}/oauth2/default`,
```
but I think it would be better as
```js
baseUrl: `https://${process.env.OKTA_DOMAIN}/oauth2/default`,
```
